docs/source/java/modules.rst

```diff
@@ -0,0 +1,413 @@
+
+.. list-table:: Arrow Java Implementations and Bindings
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Module
+     - Root Decision
+   * - Arrow Memory
+     - Native implementation.
+   * - Arrow Vector
+     - Native implementation.
+   * - Arrow Flight Grpc
+     - Native implementation.
+   * - Arrow Flight Sql
+     - Native implementation.
+   * - Arrow Algorithm
+     - Native implementation.
+   * - Arrow Compression
+     - Native implementation.
+   * - Arrow C Data Interface
+     - Native implementation.
+   * - Arrow Dataset
+     - Bindings
+   * - Arrow ORC
+     - Bindings
+   * - Arrow Gandiva
+     - Bindings
+
+Arrow java is divided in these modules to offer in-memory columnar data structures:
+
+.. list-table:: Arrow Java Modules
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Module
+     - Description
+   * - arrow-format
+     - Generated Java files from the IPC Flatbuffer definitions.
+   * - arrow-memory-core
+     - Core off-heap memory management libraries for Arrow ValueVectors.
+   * - arrow-memory-unsafe
+     - Allocator and utils for allocating memory in Arrow based on sun.misc.Unsafe.
+   * - arrow-memory-netty
+     - Netty allocator and utils for allocating memory in Arrow.
+   * - arrow-vector
+     - An off-heap reference implementation for Arrow columnar data format.
+   * - arrow-tools
+     - Java applications for working with Arrow ValueVectors.
+   * - arrow-jdbc
+     - (`Experimental`) A library for converting JDBC data to Arrow data.
+   * - arrow-plasma
+     - (`Experimental`) Java client for the Plasma object store.
+   * - flight-core
+     - (`Experimental`) An RPC mechanism for transferring ValueVectors.
+   * - flight-grpc
+     - (`Experimental`) Contains utility class to expose Flight gRPC service and client
+   * - flight-sql
+     - (`Experimental`) Contains utility classes to expose Flight SQL semantics for clients and servers over Arrow Flight.
+   * - flight-integration-tests
+     - Integration tests for Flight RPC.
+   * - arrow-performance
+     - JMH Performance benchmarks for other Arrow libraries.
+   * - arrow-algorithm
+     - (`Experimental`) A collection of algorithms for working with ValueVectors.
+   * - arrow-avro
+     - (`Experimental`) A library for converting Avro data to Arrow data.
+   * - arrow-compression
+     - (`Experimental`) A library for working with the compression/decompression of Arrow data.
+   * - arrow-c-data
+     - Java implementation of C Data Interface
+   * - arrow-orc
+     - (`Experimental`) A JNI wrapper for the C++ ORC reader implementation.
+   * - arrow-gandiva
+     - Java wrappers around the native Gandiva SQL expression compiler.
+   * - arrow-dataset
+     - Java implementation of Arrow Dataset API/Framework
+
+For more detail about how to install this modules please review
+:doc:`Installing Java Modules <install>`.
+
+.. note::
+
+    Arrow java modules offer support to work data: (1) in-memory,
+    (2) at rest and (3) on-the-wire.
```

Comment:
I'm not sure this is useful as a note. This can be a brief sentence in either the modules overview or the getting started guide.

Reply:
Updated